src/site/stages/build/plugins/create-drupal-debug.js

```diff
@@ -1,29 +1,57 @@
 /* eslint-disable no-param-reassign */
 
+const ENVIRONMENTS = require('../../../constants/environments');
 const { ENABLED_ENVIRONMENTS } = require('../../../constants/drupals');
 const { logDrupal: log } = require('../drupal/utilities-drupal');
 
+function createErrorPage(drupalError) {
+  return `
+    <!doctype html>
+    <html>
+    <body>
+      <h1>An error occurred while executing the Metalsmith-Drupal plugin</h1>
+      <pre>${JSON.stringify(drupalError)}</pre>
+    </body>
+    </html>
+  `;
+}
+
+function createIndexPage(files) {
+  const drupalPages = Object.keys(files)
+    .filter(fileName => files[fileName].isDrupalPage)
+    .map(fileName => `<li><a href="/${fileName}">${fileName}</a></li>`)
+    .join('');
+
+  const drupalIndex = `
+    <h1>The following pages were provided by Drupal:</h1>
+    <ol>${drupalPages}</ol>
+  `;
+
+  return drupalIndex;
+}
+
 function createDrupalDebugPage(buildOptions) {
-  if (!ENABLED_ENVIRONMENTS.has(buildOptions.buildtype)) {
+  // Disable this page if Drupal is not promoted to this environment, or if we're in production, because
+  // a debug page should never be built there.
+  if (
+    !ENABLED_ENVIRONMENTS.has(buildOptions.buildtype) ||
+    buildOptions.buildtype === ENVIRONMENTS.VAGOVPROD
+  ) {
     const noop = () => {};
     return noop;
   }
 
   return (files, smith, done) => {
     log('Drupal debug page written to /drupal/debug.');
 
-    const drupalPages = Object.keys(files)
-      .filter(fileName => files[fileName].isDrupalPage)
-      .map(fileName => `<li><a href="/${fileName}">${fileName}</a></li>`)
-      .join('');
+    let drupalDebugPage = null;
+    const { drupalError } = smith.metadata();
 
-    const drupalIndex = `
-      <h1>The following pages were provided by Drupal:</h1>
-      <ol>${drupalPages}</ol>
-    `;
+    if (drupalError) drupalDebugPage = createErrorPage(drupalError);
```

Comment:
Maybe it's just personal preference, but removing braces on if/else statements makes me nervous. Easy to mess up later. `const drupalDebugPage = drupalError ? createErrorPage(drupalError) : reateIndexPage(files);` is also pretty succinct.

Reply:
Switched, but Prettier broke it into separate lines 😞

Reply:
I prefer that, actually. :)